syntax = "proto2";

package CertainPB;

message EntryRecord
{
    optional uint32 prepared_num = 1;
    optional uint32 promised_num = 2;
    optional uint32 accepted_num = 3;

    optional uint64 value_id = 4;
    repeated uint64 value_uuid = 5;
    optional bytes value = 6;

    optional bool chosen = 7;
}

message CmdHeader
{
    optional uint64 uuid = 1;

    optional uint64 entity_id = 2;
    optional uint64 entry = 3;
    optional bool quick_rsp = 4;
}

message PaxosCmd
{
    optional CmdHeader header = 1;

    optional uint32 src_acceptor_id = 2;
    optional uint32 dest_acceptor_id = 3;

    optional EntryRecord src_record = 4;
    optional EntryRecord dest_record = 5;

    optional uint64 max_chosen_entry = 6;

    optional bool check_empty = 7;

    optional int32 result = 8;
}

message SimpleCmd
{
    optional CmdHeader header = 1;

    optional uint32 sub_cmd_id = 2;
    optional bytes key = 3;
    optional bytes value = 4;

    optional int32 result = 5;
}
